### PR TITLE
Add Rule::Other to cover newly defined flags. #682

### PR DESCRIPTION
**Pull request by garyhai**
Fix bug of unhandled newly defined flags when call `acl_getuser`.

### CODE REVIEW

**Comment by djc** [comment]
Thanks for helping out!

This technically still breaks compatibility, since it adds a variant on an enum that is not marked as non-exhaustive. That's fine, though, since the main branch already broke compatibility. Given that this is the case, I think it would make sense to also add variants for any newly defined flags that we know about at this point.

**Comment by garyhai** [comment]
> This technically still breaks compatibility, since it adds a variant on an enum that is not marked as non-exhaustive. That's fine, though, since the main branch already broke compatibility. Given that this is the case, I think it would make sense to also add variants for any newly defined flags that we know about at this point.

Instead of causing compatibility issues every time a new flag is added, Rule::Other is used as a transitional solution to ease compatibility. Rule may enumerate explicitly a list of commonly used flags.

File: redis/src/acl.rs [hunk 1]
```diff
@@ -95,6 +100,8 @@ impl ToRedisArgs for Rule {
             ResetKeys => out.write_arg(b"resetkeys"),
 
             Reset => out.write_arg(b"reset"),
+
+            Other(rule) => out.write_arg_fmt(format_args!("{}", rule)),
```
Comment (djc) [inline redis/src/acl.rs:104]: Instead of this, please use `out.write_arg(rule.as_bytes())`

File: redis/src/acl.rs [hunk 2]
```diff
@@ -162,7 +169,7 @@ impl FromRedisValue for AclInfo {
                             b"allkeys" => Ok(Rule::AllKeys),
                             b"allcommands" => Ok(Rule::AllCommands),
                             b"nopass" => Ok(Rule::NoPass),
-                            _ => Err(not_convertible_error!(flag, "Expect a valid ACL flag")),
+                            other => Ok(Rule::Other(String::from_utf8_lossy(other).to_string())),
```
Comment (djc) [inline redis/src/acl.rs:172]: Please use `into_owned()` instead of `to_string()` here.

Comment (garyhai) [reply redis/src/acl.rs:172]: Thanks! This is what I call professionalism.

**Review by djc — APPROVED** [review]
This looks good to me. @jaymell any thoughts?

**Comment by jaymell** [comment]
I can be persuaded here, but I'm not thrilled with the change. The core issue, of course, is the existing API has not left us with great options. But I wonder if, given that we're going to make a backward-incompatible change here, if we should consider marking the enum as non-exhaustive and adding in the new variants as appropriate?

**Comment by garyhai** [comment]
> I can be persuaded here, but I'm not thrilled with the change. The core issue, of course, is the existing API has not left us with great options. But I wonder if, given that we're going to make a backward-incompatible change here, if we should consider marking the enum as non-exhaustive and adding in the new variants as appropriate?

The non-exhaustion structure has too many limitations and imposes too much mental burden. Rule::Other is just for exhaustion and ergonomics. I suggest putting this break change into the next non-compatibility release.

**Comment by jaymell** [comment]
Ok, merging this. Thanks for the contribution.

**Pull request merged**